plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -98,9 +99,11 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
 
         // Note: it is important that `body` is only set after `embedUserFunction` is complete, as we need to
         // place the embedding in the map before processing the body.
-        embedUserFunction(declaration.symbol, signature).apply {
+        val userFunc = embedUserFunction(declaration.symbol, signature).apply {
             body = stmtCtx.convertMethodWithBody(declaration, signature, returnTarget)
         }
+
+        val validity = userFunc.body?.debugExpEmbedding!!.checkValidity()
```

Comment:
1. Please don't use `debugExpEmbedding` for non-debugging purposes.
2. Make sure to use `validity` somehow.